libglusterfs/src/statedump.c

```diff
@@ -330,54 +353,51 @@ gf_proc_dump_mem_info_to_dict(dict_t *dict)
 {
     if (!dict)
         return;
-#ifdef HAVE_MALLINFO
-    struct mallinfo info;
+#if defined(HAVE_MALLINFO2) || defined(HAVE_MALLINFO)
     int ret = -1;
 
-    memset(&info, 0, sizeof(struct mallinfo));
-    info = mallinfo();
+    GF_MALLINFO();
 
-    ret = dict_set_int32(dict, "mallinfo.arena", info.arena);
+    ret = GF_DICT_MINFO_VALUE(dict, "mallinfo.arena", info.arena);
```

Comment:
These values are read in some other places. We need to use `dict_get_uint64()` there, at least when mallinfo2 is used, though I think we could always use uint64 unconditionally.

Reply:
Maybe a bit overkill for 32-bit system but OK anyway.